internal/analytics/mock/analytics.go

```diff
@@ -0,0 +1,46 @@
+package mock
```

Comment:
Please don't use mocks. Either the functions we're testing need to be written in such a way that they don't rely on functionality we're not interested in testing, or that functionality should just have sensible defaults that don't throw a wrench into the mix.

IIRC our analytics code had a check in there to see if we were in tests, and if so it didn't fire events. I'd still favour that over us writing and maintaining mocks.

Consider mocks banned from our codebase, unless we agree as a team that they are feasible for a particular use-case.